File: backend/src/main/java/com/sportsstore/infrastructure/persistence/mapper/ProductMapper.java
package com.sportsstore.infrastructure.persistence.mapper;

import com.sportsstore.domain.entity.Product;
import com.sportsstore.infrastructure.persistence.entity.ProductJpaEntity;
import org.springframework.stereotype.Component;

@Component
public class ProductMapper {

    public ProductJpaEntity toJpaEntity(Product product) {
        ProductJpaEntity entity = new ProductJpaEntity();
        entity.setId(product.getId());
        entity.setName(product.getName());
        entity.setDescription(product.getDescription());
        entity.setPrice(product.getPrice());
        entity.setOriginalPrice(product.getOriginalPrice());
        entity.setStockQuantity(product.getStockQuantity());
        entity.setSku(product.getSku());
        entity.setActive(product.isActive());
        entity.setImages(product.getImages());
        entity.setCategories(product.getCategories());
        entity.setBrand(product.getBrand());
        entity.setCreatedAt(product.getCreatedAt());
        entity.setUpdatedAt(product.getUpdatedAt());
        return entity;
    }

    public Product toDomainEntity(ProductJpaEntity entity) {
        return new Product.Builder()
                .id(entity.getId())
                .name(entity.getName())
                .description(entity.getDescription())
                .price(entity.getPrice())
                .originalPrice(entity.getOriginalPrice())
                .stockQuantity(entity.getStockQuantity())
                .sku(entity.getSku())
                .active(entity.isActive())
                .images(entity.getImages())
                .categories(entity.getCategories())
                .brand(entity.getBrand())
                .build();
    }
}

File: frontend/src/app/infrastructure/mappers/product.mapper.ts
import { Injectable } from '@angular/core';
import { Product } from '../../domain/entities/product.entity';

@Injectable({
  providedIn: 'root'
})
export class ProductMapper {
  toDomain(dto: any): Product {
    return new Product(
      dto.id,
      dto.name,
      dto.description,
      dto.price,
      dto.originalPrice,
      dto.stockQuantity,
      dto.sku,
      dto.active,
      dto.images || [],
      dto.categories || [],
      dto.brand,
      new Date(dto.createdAt),
      new Date(dto.updatedAt)
    );
  }

  toDto(product: Product): any {
    return {
      name: product.name,
      description: product.description,
      price: product.price,
      originalPrice: product.originalPrice,
      stockQuantity: product.stockQuantity,
      sku: product.sku,
      brand: product.brand,
      categories: product.categories,
      images: product.images
    };
  }
}
